test/functional/helpers.lua

```diff
@@ -773,7 +773,7 @@ function module.missing_provider(provider)
     return prog == '' and (provider .. ' not detected') or false
   elseif provider == 'python' or provider == 'python3' then
     local py_major_version = (provider == 'python3' and 3 or 2)
-    local errors = module.funcs['provider#pythonx#Detect'](py_major_version)[2]
+    local errors = module.funcs['provider#pythonx#Detect'](py_major_version)[3]
```

Comment:
Because `provider#pythonx#Detect` now returns the full path to the provider interpreter, **its version**, and only then any errors.